torchrl/data/datasets/gen_dgrl.py

```diff
@@ -0,0 +1,367 @@
+        datasets = [
+            "bigfish",
+            "bossfight",
+            "caveflyer",
+            "chaser",
+            "climber",
+            "coinrun",
+            "dodgeball",
+            "fruitbot",
+            "heist",
+            "jumper",
+            "leaper",
+            "maze",
+            "miner",
+            "ninja",
+            "plunder",
+            "starpilot",
+        ]
+        categories = [
+            "1M_E",
+            "1M_S",
+            "10M",
+            "25M",
+            "level_1_E",
+            "level_1_S",
+            "level_40_E",
+            "level_40_S",
+        ]
+
+        return ["-".join((ds, cat)) for cat in categories for ds in datasets]
+
+    def __init__(
+        self,
+        dataset_id: str,
+        batch_size: int = None,
+        *,
+        download: bool = True,
+        root: str | None = None,
+        **kwargs,
+    ):
+        self.dataset_id = dataset_id
+        try:
+            dataset, category = dataset_id.split("-")
+        except Exception:
+            category = dataset_id
+            dataset = None
+        self._dataset_name = dataset
+        self._category_name = category
+        if root is None:
+            root = _get_root_dir("gen_dgrl")
+            os.makedirs(root, exist_ok=True)
+        self.root = root
+        if download == "force" or (download and not self._is_downloaded()):
+            storage = TensorStorage(self._download_and_preproc())
+        else:
+            storage = TensorStorage(TensorDict.load_memmap(self.data_path_root))
+        super().__init__(storage=storage, batch_size=batch_size, **kwargs)
+
+    @property
+    def data_path(self):
+        if self.split_trajs:
+            return Path(self.root) / (self.dataset_id + "_split")
+        return self.data_path_root
+
+    @property
+    def data_path_root(self):
+        return Path(self.root) / self.dataset_id
+
+    def _is_downloaded(self):
+        return os.path.exists(self.data_path_root)
+
+    def _download_and_preproc(self):
+        dataset, category = self._dataset_name, self._category_name
+        link = self._build_urls_with_category_name(dataset, category)
+        data_link = (category, self._fetch_file_name_from_link(link), link)
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            self._download_category_file(
+                tmpdir, skip_downloaded_files=True, link=data_link
+            )
+            return self._unpack_category_file(
+                tmpdir, clear_archive=True, link=data_link, category_name=category
+            )
+
+    @classmethod
+    def _build_urls_with_category_name(
+        cls, dataset, category_name: str
+    ) -> tp.List[str]:
+        path = [cls.BASE_URL, cls._convert_category_name(category_name)]
+        path += [f"{dataset}.tar.xz"]
+        return os.path.join(*path)
+
+    @staticmethod
+    def _convert_category_name(category_name: str) -> str:
+        if category_name == "1M_E":
```

Comment:
This is good for now. In the future if we have more datasets, we would want to modify the `category_name` to reflect the parameters, like "1M_200_E", "100k_40_S", etc, so that we can easily map them separately and combine together for paths:
- "1M" -> "1M", "100k" -> "100k", ...
- "200" -> "level_200", "40" -> "level_40", ...
- "E" -> "expert", "S" -> "suboptimal"

Reply:
Sure! If you add more of these just ping us and we'll do the fixes